### PR TITLE
Master mail owl chatter xdu

### PR DESCRIPTION
**Pull request by xdu-odoo**
- First part of chatter (opw 2089858)
- attachment style (opw 2089878)
- thread in desc order (opw 2089876)
- attachment box (opw 2089870)

### CODE REVIEW

**Comment by xdu-odoo** [comment]
@alexkuhn I rebased on master-mail-owl but I didn't take the _dragCount part you (or seb) did on composer because it wasn't working with the hook I created.

And as I don't understand what (and why) you are doing I preferred to stick to what is working on my branch ; so maybe the hook and the composer need to be changed but I don't know how.

~I still need to change threadLocalId to threadLocalds on attachments.~ (Edit : it's done with last commit)

**Comment by xdu-odoo** [comment]
I managed to bring back your logic (I think), the tests are green.
It would be great to merge this ASAP because every change made in the core branch could be a pain to reflect in new files ...

File: addons/mail/static/src/js/owl/component/message_list.js [hunk 1]
```diff
@@ -336,11 +338,19 @@ class MessageList extends Component {
...
+            return (
+                // loadMore top is upper than message list bottom (modulo offset)
+                // AND loadMore bottom is lower than message lit top (modulo offset)
+                loadMoreRect.top < elRect.bottom + offset &&
+                loadMoreRect.bottom > elRect.top - offset
+            );
+        } else {
+            // loadMore bottom is upper than message list bottom (modulo offset)
+            return loadMoreRect.top < elRect.bottom + offset;
+        }
```
Comment (alexkuhn) [inline addons/mail/static/src/js/owl/component/message_list.js]: "visibility" of loadMore is not order-dependent, so I think this code should neither.

Comment (xdu-odoo) [reply addons/mail/static/src/js/owl/component/message_list.js]: I think it is as the order determines the fact that the loadmore is on the top or in the bottom of the message list ... so in a direction you want the loadmore button to be "below" the top of the message list container so it is visible and in the other one you want the loadmore button be "above" the bottom of the message list container

Comment (xdu-odoo) [reply addons/mail/static/src/js/owl/component/message_list.js]: simplified :)

**Comment by xdu-odoo** [comment]
Runbot said :

> "QUnit test failed: "mail.owl > component > ChatWindowManager > chat window: state destroyed on close" [message: "Cannot read property 'attachmentLocalIds' of undefined", actual: "TypeError: Cannot read property 'attachmentLocalIds' of undefined", expected: "undefined"]

I cannot reproduce that locally even by launching js tests via the python launcher...

**Pull request merged**